hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -1724,62 +1993,40 @@ field_kind_based_operations::field_kind_based_operations(
   : kind_size_in_byte_func(std::move(kind_size_in_byte_func))
 {}
 field_kind_based_operations
-field_operations::get(enum field_kind field_kind)
-{
-    static const field_kind_based_operations ALL[NUMBER_OF_FIELD_KINDS] = {
-        field_kind_based_operations{ []() { return 0; } },
-        field_kind_based_operations{},
-        field_kind_based_operations{ []() { return 1; } },
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{
-          []() { return util::Bits::SHORT_SIZE_IN_BYTES; } },
-        field_kind_based_operations{},
-        field_kind_based_operations(
-          []() { return util::Bits::INT_SIZE_IN_BYTES; }),
-        field_kind_based_operations{},
-        field_kind_based_operations{
-          []() { return util::Bits::LONG_SIZE_IN_BYTES; } },
-        field_kind_based_operations{},
-        field_kind_based_operations{
-          []() { return util::Bits::FLOAT_SIZE_IN_BYTES; } },
-        field_kind_based_operations{},
-        field_kind_based_operations{
-          []() { return util::Bits::DOUBLE_SIZE_IN_BYTES; } },
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{},
-        field_kind_based_operations{}
-    };
-    return ALL[field_kind];
+field_operations::get(field_kind kind)
+{
+    using util::Bits;
```

Comment:
I think using a static array is better in terms of performance

Reply:
maybe we can also use templates and explicity specialize for BOOLEAN, INT8, INT16, INT32, INT64, FLOAT32 and FLOAT64.